### PR TITLE
Fixing TCMalloc mapped pages bug in fixed heap monitor

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: A previous PR [22585](https://github.com/envoyproxy/envoy/pull/22585) attempts to fix a bug from prior but was not implemented correctly with the runtime guard. The legacy code was still being run by default. This PR moves the code around to create new default behavior.
Risk Level: Low
Testing: Fixed tests to use proper runtime flags

### CODE REVIEW

File: changelogs/current.yaml [hunk 1]
```diff
@@ -55,6 +55,12 @@ minor_behavior_changes:
     the cookie is still accepted, but is planned to be obsoleted in the future.
     This behavior change can be reverted by setting
     ``envoy.reloadable_features.stateful_session_encode_ttl_in_cookie`` to ``false``.
+- area: resource_monitors
+  change: |
+    Changed behavior of the fixed heap monitor to count pages allocated to
+    TCMalloc as free memory if it's not used by Envoy. This change can be
```
Comment (KBaichoo) [inline changelogs/current.yaml:61]: s/count pages allocated to TCMalloc as free memory if it's not used by Envoy./count unused mapped pages as free memory./

File: source/extensions/resource_monitors/fixed_heap/fixed_heap_monitor.cc [hunk 2]
```diff
@@ -27,18 +27,19 @@ FixedHeapMonitor::FixedHeapMonitor(
 void FixedHeapMonitor::updateResourceUsage(Server::ResourceUpdateCallbacks& callbacks) {
 
   auto computeUsedMemory = [this]() -> size_t {
+    // TODO(Diazalan): Remove if statement once runtime guard is deprecated
     if (!Runtime::runtimeFeatureEnabled(
```
[the remark this answers is not in the document]
Comment (DiazAlan) [reply source/extensions/resource_monitors/fixed_heap/fixed_heap_monitor.cc:31]: Done

File: changelogs/current.yaml (hunk 1 above)
Comment (DiazAlan) [reply changelogs/current.yaml]: Done

File: source/common/runtime/runtime_features.cc [hunk 3]
```diff
@@ -32,8 +32,8 @@
 RUNTIME_GUARD(envoy_reloadable_features_allow_compact_maglev);
 RUNTIME_GUARD(envoy_reloadable_features_append_query_parameters_path_rewriter);
 RUNTIME_GUARD(envoy_reloadable_features_conn_pool_delete_when_idle);
+RUNTIME_GUARD(envoy_reloadable_features_count_mapped_pages_as_free);
```
Comment (KBaichoo) [inline source/common/runtime/runtime_features.cc:35]: this is a bit ambiguous; maybe `count_unused_mapped_pages_as_free` ?

**Review by KBaichoo — COMMENTED** [review]
lgtm otherwise, thank you

File: source/common/runtime/runtime_features.cc (hunk 3 above)
Comment (DiazAlan) [reply source/common/runtime/runtime_features.cc]: Done

**Review by KBaichoo — APPROVED** [review]

**Pull request merged**